src/lib/ReactCore_Internal.re

```diff
@@ -812,23 +894,26 @@ module Make = (Implementation: HostImplementation) => {
      * Rendering produces a list of instance trees.
      */
     type t = renderedElement;
-    let listToRenderedElement = (renderedElements) =>
+    let listToRenderedElement = renderedElements =>
       INested("List", renderedElements);
-    let render = (reactElement) : t => Render.renderReactElement(reactElement);
-    let update = (renderedElement: t, reactElement) : (t, UpdateLog.t) => {
+    let render = reactElement : t => Render.renderReactElement(reactElement);
+    let update = (renderedElement: t, reactElement) : (t, option(UpdateLog.topLevelUpdate)) => {
       let updateLog = UpdateLog.create();
-      let newRenderedElement =
+      let (topLevelChange, newRenderedElement) =
         Render.updateRenderedElement(
-          ~topLevelUpdate=true,
           ~updateLog,
           (renderedElement, reactElement, reactElement)
         );
-      (newRenderedElement, updateLog)
+      switch topLevelChange {
+      | NoChange => (renderedElement, None)
+      | x => (newRenderedElement, Some({typ: x, updateLog}))
+      };
     };
 
     /***
      * Flush the pending updates in an instance tree.
      * TODO: invoke lifecycles
+     * TODO: (Maybe) return topLevelChange from here
```

Comment:
No, `topLevelChange` should not be returned from here since the changes are happening within the tree.